python-package/lightgbm/compat.py

```diff
@@ -63,7 +63,6 @@ class DataFrame(object):
     from sklearn.base import BaseEstimator
     from sklearn.base import RegressorMixin, ClassifierMixin
     from sklearn.preprocessing import LabelEncoder
-    from sklearn.utils import deprecated
```

Comment:
why did you remove `deprecated`?

Reply:
It was the reason of failing on Python 2.
I replaced it with `warnings.warn('...', LGBMDeprecationWarning)` which is more universal mechanism and doesn't depend on sklearn.

Reply:
I don't know why `deprecated` doesn't work with Python 2... even though scikit-learn supports Python 2.

Reply:
Yeah, it was strange for me too. I suppose it happens because `Mock()` didn't applied properly to this decorator in Python 2. Nevertheless, now we get rid of this headache.